pymatgen/io/openmm/inputs.py

```diff
@@ -0,0 +1,301 @@
+        return TopologyInput(topology, positions)
+
+
+class SystemInput(InputFile):
+    def __init__(self, system: System):
+        self.system = system
+
+    def get_string(self) -> str:
+        return XmlSerializer.serialize(self.system)
+
+    @classmethod
+    def from_string(cls, contents: str):
+        return SystemInput(XmlSerializer.deserialize(contents))
+
+
+class IntegratorInput(InputFile):
+    def __init__(self, integrator: Integrator):
+        self.integrator = integrator
+
+    def get_string(self) -> str:
+        return XmlSerializer.serialize(self.integrator)
+
+    @classmethod
+    def from_string(cls, contents: str):
+        return IntegratorInput(XmlSerializer.deserialize(contents))
+
+
+class StateInput(InputFile):
+    def __init__(self, state: State):
+        self.state = state
+
+    def get_string(self) -> str:
+        return XmlSerializer.serialize(self.state)
+
+    @classmethod
+    def from_string(cls, contents: str):
+        return StateInput(XmlSerializer.deserialize(contents))
+
+
+class OpenMMSet(InputSet):
+
+    # TODO: if there is an optional file should it be missing or should the value be none?
+    @classmethod
+    def from_directory(
+        cls,
+        directory: Union[str, Path],
+        topology_file: str = "topology.pdb",
+        system_file: str = "system.xml",
+        integrator_file: str = "integrator.xml",
+        state_file: str = "state.xml",
+    ):
+        topology = TopologyInput.from_file(topology_file)
+        system = TopologyInput.from_file(system_file)
+        integrator = TopologyInput.from_file(integrator_file)
+        openmm_set = OpenMMSet(
+            topology=topology,
+            system=system,
+            integrator=integrator,
+        )
+        if Path("state.xml").is_file():
+            state = TopologyInput.from_file(state_file)
+            openmm_set["state"] = state
+        return openmm_set
+
+    def validate(self) -> bool:
+        # TODO: this should test if the set returns a valid simulation and throw an error if it does not
+        # TODO: is this condition too strict?
+        try:
+            self.get_simulation()
+        except Exception as e:
+            print(
+                "A valid simulation could not be generated, the following error was raised:",
+                e,
+            )
+            return False
+        else:
+            return True
+
+    def get_simulation(self) -> Simulation:
+        simulation = Simulation(
+            self.topology.topology,
+            self.system.system,
+            self.integrator.integrator,
+        )
+        if hasattr(self, "state") and self.state:
+            # TODO: confirm that this works correctly
+            simulation.context.setState(self.state.state)
+        return simulation
+
+
+# noinspection PyMethodOverriding
+class OpenMMGenerator(InputGenerator):
+
+    # TODO: what determines if a setting goes in the __init__ or get_input_set?
```

Comment:
@orioncohen I would say anything that is _system specific_ goes to `get_input_set` (so typically, just your coordinates) whereas all other settings e.g. temperature, run time, integrator, etc. are part of the generator. I think of the generator as the "recipe" for a particular calculation.

The distinction for MD is admittedly murkier than it is for other types of calculations. For example. I could be convinced that basic things such as step size and run time ought to go into `get_input_set` since they don't have any bearing on the actual content of the calculation (forcefield, etc.). What do you think?

Reply:
> 
> 
> What should determine if a parameter goes in `InputGenerator.__init__` or `InputGenerator.get_input_set`?
> 
> The documentation seems to indicate that only the structure and other critical parameters should go in `get_input_set`, but I can imagine many situations where that would be inconvenient. For example, if I wanted to repeat a simulation at four different temperatures. I'm tempted to put all simulations settings in both `__init__` and `get_input_set` for maximum flexibility, but I'm curious to hear more about the intended design.

Personally I am reluctant to duplicate settings in both `__init__` and `get_input_set` . I feel we have a lot of that kind of duplication in the existing pymatgen-custodian-atomate VASP infrastructure and it makes it very difficult to track down where things are being set. I'd prefer you divide them up so that there is only one place to set each thing. We can revisit this decision if it becomes too cumbersome in practice though.

I agree with you that in this context, parameters like temperature and pressure would make sense in `get_input_set`. The generator could define the force field, the integrator, and the fact that this is (for example) an NPT simulation. Specific instances of that recipe can change the Temp or Pressure without fundamentally changing the technical decisions that constitute the recipe for the simulation.

Reply:
That makes sense to me. In some of my use cases, I'd have to define many OpenMM generators, but I'd say its worth the clarity.

Reply:
> 
> 
> That makes sense to me. In some of my use cases, I'd have to define many OpenMM generators, but I'd say its worth the clarity.

I think that's OK, maybe even what we're going for in the generator approach? Since the generators don't store structure, they are lightweight objects that are intended to be customized a lot. The intent is for all the customization to happen there rather than down in `get_input_set`.

I think we'll have to see how this "feels" in practice in order to find the right balance of where to put different settings

Reply:
Got it. I'll keep this design principle in mind as I finish up the IO.

Reply:
One thing that *might* belong in `get_input_set` is the `Platform` and `platformProperties` arguments, which are specific to the HPC resource. `Platform` species whether CUDA, OpenCL, or CPU will be used and `platformProperties` can specify properties like GPU device index and number of threads.